src/js/modules/RoutingApp.js

```diff
@@ -25,195 +26,136 @@ import {
-                      exact
-                      path={`/about/${description.slug}`}
-                      key={`description${description.id}`}
-                      render={props => (
-                        <DescriptionPage description={description} />
-                      )}
-                    />
-                  );
-                })}
-                <Route
-                  exact
-                  path={"/contributors/:contributor_slug"}
-                  key={"contributors"}
-                  render={props => <ContributorPage match={props.match} />}
-                />
-                <Route
-                  exact
-                  path={"/illustrators/:illustrator_slug"}
-                  key={"illustrators"}
-                  render={props => <IllustratorPage match={props.match} />}
-                />
-                <Route
-                  exact
-                  path={"/photographers/:photographer_slug"}
-                  key={"photographers"}
-                  render={props => <PhotographerPage match={props.match} />}
-                />
-                <Route
-                  exact
-                  path={"/myaccount"}
-                  key={"signIn"}
-                  render={() =>
-                    session ? (
-                      <Redirect to="/myaccount/profile" />
-                    ) : (
-                      <SignInPage />
-                    )}
-                />
-                <Route
-                  exact
-                  path="/myaccount/sign-up"
-                  key={"signUp"}
-                  render={() =>
-                    session ? (
-                      <Redirect to="/myaccount/profile" />
-                    ) : (
-                      <SignUpPage />
-                    )}
-                />
-                <Route
-                  exact
-                  path="/myaccount/profile"
-                  key={"profile"}
-                  render={() =>
-                    session ? <ProfilePage /> : <Redirect to="/myaccount" />}
-                />
+const RoutingApp = ({
+  data: { loading, error, allSections },
+  session,
+  descriptions,
+}) => {
+  return (
+    <ConnectedRouter
+      onUpdate={() => window.scrollTo(0, 0)}
+      history={appHistory}
+    >
+      <PageLayout>
+        {!loading ? (
+          <Switch>
+            <Route exact path="/" component={HomePage} />
+            {allSections.map(section => {
+              return (
                 <Route
                   exact
-                  path="/myaccount/profile/edit"
-                  key={"editProfile"}
-                  render={() =>
-                    session ? (
-                      <EditProfilePage />
-                    ) : (
-                      <Redirect to="/myaccount" />
-                    )}
+                  path={section.permalink}
+                  key={`section${section.id}`}
+                  render={props => <SectionPage section={section} />}

```

Comment:
Actually...yeah same deal. Why bother giving SectionPage section? It's going to have to do a massive query anyways. It's not a big deal, but like...why? Also props is unused, just do `() =>`

Reply:
I need this so I don't think `/section/subsection` is the same Route as `/section/article`.